File: java/trip-service-kata/src/main/java/org/craftedsw/tripservicekata/user/UserSession.java
package org.craftedsw.tripservicekata.user;

import org.craftedsw.tripservicekata.exception.DependendClassCallDuringUnitTestException;

public class UserSession {

	private static final UserSession userSession = new UserSession();
	
	private UserSession() {
	}
	
	public static UserSession getInstance() {
		return userSession;
	}

	public boolean isUserLoggedIn(User user) {
		throw new DependendClassCallDuringUnitTestException(
				"UserSession.isUserLoggedIn() should not be called in an unit test");
	}

	public User getLoggedUser() {
		throw new DependendClassCallDuringUnitTestException(
				"UserSession.getLoggedUser() should not be called in an unit test");
	}

}

File: kotlin/trip-service-kata/src/main/kotlin/org/craftedsw/tripservicekata/user/UserSession.kt
package org.craftedsw.tripservicekata.user

import org.craftedsw.tripservicekata.exception.CollaboratorCallException

class UserSession {

    private constructor()

    companion object {
        @JvmStatic val instance = UserSession()
    }

    val loggedUser: User?
        get() = throw CollaboratorCallException("UserSession.getLoggedUser() should not be called in an unit test")

}
